cmd/do-agent/config.go

```diff
@@ -284,3 +294,20 @@ func disableCollectors(names ...string) {
 func disableCollectorFlag(name string) string {
 	return fmt.Sprintf("--no-collector.%s", name)
 }
+
+func convertToLabelPairs(s []string) []*dto.LabelPair {
+	var l []*dto.LabelPair
+	for _, lbl := range s {
+		vals := strings.SplitN(lbl, ":", 2)
+		if len(vals) != 2 { // require a key value pair
+			log.Fatal("Bad additional-label %s, must be in the format of <key>:<value>", lbl)
+		}
+
+		l = append(l, &dto.LabelPair{
+			Name:  &vals[0],
+			Value: &vals[1],
```

Comment:
URN's normally have ":" in them.  We should possibly use a better delimiter than ":" for this reason, incase we wan't to use this to attach urns at some point.

Reply:
nvm, I see you used SplitN above

Reply:
well we only split after the first delimiter

Reply:
so something like resource:my:urn:fields will still work